components/support/locale/src/main/java/mozilla/components/support/locale/LocaleManager.kt

```diff
@@ -48,6 +50,34 @@ object LocaleManager {
         return currentLocale
     }
 
+    /**
+     * Change the current locale to the system defined one.
+     * [getCurrentLocale] will return null
+     *
+     * After calling this function, to visualize the locale changes you have to make sure all your visible activities
+     * get recreated. If your app is using the single activity approach, this will be trivial just call
+     * [AppCompatActivity.recreate]. On the other hand, if you have multiple activity this could be tricky, one
+     * alternative could be restarting your application process see https://github.com/JakeWharton/ProcessPhoenix
+     *
+     */
+    fun resetToSystemDefault(context: Context) {
+        clear(context)
+        val locale = getSystemDefault()
+        locale?.let {
+            updateSystemLocale(locale)
+            updateConfiguration(context, locale)
+        }
+    }
+
+    /**
+     * Returns the locale set by the system
+     */
+    @Suppress("DEPRECATION")
+    fun getSystemDefault(): Locale? {
+        val config = Resources.getSystem().configuration
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) config.locales.get(0) else config.locale
```

Comment:
Maybe [ConfigurationCompat.getLocales ](https://developer.android.com/reference/androidx/core/os/ConfigurationCompat) could safe us the API checking :)

Reply:
Yep, great find! Changed to this compat class